server/common/dao/src/main/java/org/kaaproject/kaa/server/common/dao/service/EndpointServiceImpl.java

```diff
@@ -326,6 +326,9 @@ public EndpointProfileDto attachEndpointToUser(String userExternalId, String ten
             endpointIds = new ArrayList<>();
             endpointUser.setEndpointIds(endpointIds);
         }
+        else if (endpointIds != null && endpointIds.contains(profile.getId())) {
+            LOG.warn("Endpoint is already assigned to current user {}. Unassign it first!.", profile.getEndpointUserId());
```

Comment:
LOG.warn("Endpoint is already assigned to current user {}!", profile.getEndpointUserId());